test/e2e/page-objects/login.js

```diff
@@ -20,11 +30,56 @@ const LoginPage = Object.create(BasePage, {
       return $('input[type=submit]');
     },
   },
+  register: {
+    get() {
+      return $('.cd-login__register');
+    },
+  },
+  forgotPassword: {
+    get() {
+      return $('.cd-login__forgot-password');
+    },
+  },
+  registerLink: {
+    get() {
+      return this.register.$('a');
+    },
+  },
+  forgotPasswordLink: {
+    get() {
+      return this.forgotPassword.$('a');
+    },
+  },
+  emailReqError: {
+    get() {
+      return $('.cd-login__email-req-err');
+    },
+  },
+  emailFormatError: {
+    get() {
+      return $('.cd-login__email-format-err');
+    },
+  },
+  passwordReqError: {
+    get() {
+      return $('.cd-login__password-req-err');
+    },
+  },
+  loginFailed: {
+    get() {
+      return $('.cd-login__login-failed');
+    },
+  },
   open: {
     value() {
       return BasePage.open.call(this, '/v2/login');
     },
   },
+  openWithReferrer: {
```

Comment:
A couple things about this:

1. It won't open the login page with a referrer query param, but will open the referrer directly. You need BasePage.open to be called with `'/v2/login?referrer=%2Fdojos%2Fie%2Fsome%2Fslug'`, but you don't want `'/v2/login?referrer='` when there's no referrer given.
2. I'd actually be inclined to add this to BasePage.open as a generic handler for query parameters. Then open in this file can just take in a query object and pass it down to BasePage.open. If this doesn't make sense to you, grab me and I'll run through it with you.